### PR TITLE
Make ODF Console a Monorepo, make OCS an internal package

### PR DESCRIPTION
**Pull request by bipuladh**
/assign @cloudbehl 
```
packages
├── ocs
│   ├── OWNERS
│   └── package.json
├── odf
│   ├── components
│   │   ├── odf-dashboard
│   │   │   ├── activity-card
│   │   │   │   ├── activity-card.scss
│   │   │   │   └── activity-card.tsx
│   │   │   ├── dashboard.tsx
│   │   │   ├── object-storage-card
│   │   │   │   └── capacity-card.tsx
│   │   │   ├── performance-card
│   │   │   │   ├── performance-card.scss
│   │   │   │   ├── performance-card.tsx
│   │   │   │   └── utils.ts
│   │   │   ├── queries.ts
│   │   │   ├── status-card
│   │   │   │   ├── status-card.tsx
│   │   │   │   └── storage-system-popup.tsx
│   │   │   ├── system-capacity-card
│   │   │   │   └── capacity-card.tsx
│   │   │   └── utils.ts
│   │   └── utils
│   │       ├── common.ts
│   │       └── index.ts
│   ├── constants
│   │   ├── common.ts
│   │   └── index.ts
│   ├── features.ts
│   ├── models
│   │   ├── common.ts
│   │   ├── index.ts
│   │   ├── ocs.ts
│   │   └── odf.ts
│   ├── OWNERS
│   ├── package.json
│   ├── queries
│   │   ├── dashboard.ts
│   │   └── index.ts
│   ├── style.scss
│   └── types
│       ├── common.ts
│       ├── console-types.ts
│       └── index.ts
└── shared
    ├── dashboards
    │   ├── capacity-card
    │   │   ├── capacity-card.scss
    │   │   └── capacity-card.tsx
    │   └── line-graph
    │       ├── line-graph.scss
    │       └── line-graph.tsx
    ├── generic
    │   └── Error.tsx
    ├── heading
    │   └── page-heading.tsx
    ├── hooks
    │   └── ref-width.ts
    ├── OWNERS
    ├── package.json
    ├── popup
    │   ├── status-popup.scss
    │   └── status-popup.tsx
    ├── resource-link
    │   └── resource-link.tsx
    ├── styles
    │   ├── _common.scss
    │   └── _skeleton.scss
    ├── table
    │   └── table.tsx
    ├── types
    │   ├── common.ts
    │   └── index.ts
    └── utils
        ├── dashboard.ts
        ├── humanize.js
        └── index.ts

27 directories, 51 files
```
Description of changes:
1) Create packages for `shared` components. These components are used for creating dashboards for ODF, OCS and in future ACM DR.
2) Separation of packages to allow OCS to extend ODF dashboard to fix issues such as https://bugzilla.redhat.com/show_bug.cgi?id=2022693
3) Created an interim package for OCS which will house all OCS related components.

### CODE REVIEW

File: packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx [hunk 1]
```diff
@@ -35,7 +36,12 @@ const ObjectCapacityCard: React.FC = () => {
         </DashboardCardTitle>
       </DashboardCardHeader>
       <DashboardCardBody>
-        <CapacityCard data={dataFrames} relative isPercentage={false} />
+        <CapacityCard
+          data={dataFrames}
+          relative
+          isPercentage={false}
+          resourceModel={ODFStorageSystem}
```
Comment (afreen23) [inline packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx:43]: why this ?

**Review by afreen23 — CHANGES_REQUESTED** [review]
can you add some changelog or description , what changes are you making ?
I see some refactors which are not clear

File: packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx (hunk 1 above)
Comment (bipuladh) [reply packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx:43]: Need to do this to segregate common types and odf specific types.

**Review by afreen23 — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]

**Review by GowthamShanmugam — APPROVED** [review]

File: packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx [hunk 2]
```diff
@@ -8,8 +9,8 @@ import {
   usePrometheusPoll,
 } from '@openshift-console/dynamic-plugin-sdk-internal';
 import { useTranslation } from 'react-i18next';
-import { humanizeBinaryBytes } from '../../../humanize';
-import CapacityCard from '../../common/capacity-card/capacity-card';
+import { humanizeBinaryBytes } from '../../../../shared/utils/humanize';
```
Comment (rexagod) [inline packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx:12]: Let's leverage the different packages now for cleaner import paths (maybe replacing more than 2 `../` with their package-prefixed paths?).
```suggestion
import { humanizeBinaryBytes } from '@odf/shared/utils/humanize';
```

File: packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx [hunk 3]
```diff
@@ -8,8 +9,8 @@ import {
...
+import { ODFStorageSystem } from '../../../models';
```
Comment (rexagod) [inline packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx:13]:
```suggestion
import { ODFStorageSystem } from '@odf/core/models';
```

File: packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx [hunk 4]
```diff
@@ -13,12 +13,12 @@ import {
 } from '@openshift-console/dynamic-plugin-sdk-internal';
 import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
-import { humanizeBinaryBytes } from '../../../humanize';
-import { ODFStorageSystem } from '../../../models';
-import { StorageSystemKind } from '../../../types';
 import CapacityCard, {
   CapacityMetricDatum,
-} from '../../common/capacity-card/capacity-card';
+} from '../../../../shared/dashboards/capacity-card/capacity-card';
```
Comment (rexagod) [inline packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx:18]:
```suggestion
} from '@odf/shared/dashboards/capacity-card/capacity-card';
```

File: packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx [hunk 5]
```diff
@@ -13,12 +13,12 @@ import {
...
+import { humanizeBinaryBytes } from '../../../../shared/utils/humanize';
```
Comment (rexagod) [inline packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx:19]:
```suggestion
import { humanizeBinaryBytes } from '@odf/shared/utils/humanize';
```

File: packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx [hunk 6]
```diff
@@ -13,12 +13,12 @@ import {
...
+import { ODFStorageSystem } from '../../../models';
+import { StorageSystemKind } from '../../../types';
```
Comment (rexagod) [inline packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx:21]:
```suggestion
import { ODFStorageSystem } from '@odf/core/models';
import { StorageSystemKind } from '@odf/core/types';
```

**Review by rexagod — CHANGES_REQUESTED** [review]
This was a much needed change, thank you @bipuladh! 🎉

File: package.json [hunk 7]
```diff
@@ -106,8 +109,8 @@
       "@console/pluginAPI": "*"
     },
     "exposedModules": {
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply package.json:111]: This is an internal field that we use to expose components from our packages. Think of it like entry point of webpack.

File: packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx (hunk 2 above)
Comment (bipuladh) [reply packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx:12]: Let's do this only for cross package imports. Same package imports should use relative path.

Comment (rexagod) [reply packages/odf/components/odf-dashboard/object-storage-card/capacity-card.tsx]: Makes sense.

**Comment by rexagod** [comment]
@bipuladh Can we remove the hold now?

**Pull request merged**